### DISCUSSION TITLE
Model viewer is not getting the ar-status event

### DISCUSSION THREAD

**Discussion by bieldiaz** [google/model-viewer#3284] [Q&A]
### Description

Hello model-viewer team. I'm working with Angular 13 project. And I want to get the `ar-status` event from model viewer. But i doesn't works for me. Anyone is working with Angular and it happends the same?

I had been installed the model viewer library by npm. 

#### Live Demo
```
  const compatibleDevice$ = fromEvent(this.modelViewerElement.nativeElement, 'ar-status')
    compatibleDevice$.pipe(
      map((ev:any)=>{
        console.log(ev)
      }),
      take(1),
    ).subscribe()
```

Im getting the model-visibitilty event and this works for me very well. 

```
const loaded$ = fromEvent(this.modelViewerElement.nativeElement, 'model-visibility')
    loaded$.pipe(
      map((ev:CustomEvent)=>{
        console.log(ev)
        if(ev.detail.visible === true){
          this.isModelLoaded$.next(true)
        }else{
          this.isModelLoaded$.next(false)
        }
      }),
      take(1),
    ).subscribe()
```

### Version

- model-viewer: v1.10.1

### Browser Affected

- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [x] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [x] MacOS
- [ ] Windows

### AR

- [x] WebXR
- [x] SceneViewer
- [x] QuickLook

**Comment by samaneh-kazemi** [comment]
Hi @ bieldiaz, thanks for reaching out. I don't have much experience with Angular, but to understand the issue better, can you share more details on what happens when you use that feature?

**Comment by bieldiaz** [comment]
Hello @samaneh-kazemi, thank you for your reply. I want to detect if the device is compatible with AR with the `ar-status` event. But I see that you have to press the ar-button to know if is compatible or not. Is there any way to know before (not by clicking the ar button). Thank you very much.

**Comment by samaneh-kazemi** [comment]
Hi @benmbark, knowing if device can support AR is only approximate (you aren't allowed to know for sure due to finger-printing concerns in the browser). We hide the AR button when we know AR is not supported, but it sometimes shows when AR still fails. What problem you are trying to solve?

**Comment by iiLearner** [comment]
ar-status is not getting fired at all, is this normal?

**Comment by elalish** [comment]
ar-status only applies to WebXR AR sessions, not Scene Viewer or Quick Look (which don't give us any feedback): https://modelviewer.dev/docs/index.html#entrydocs-augmentedreality-css-arStatus

You can check if AR works (not 100%, but decent) using https://modelviewer.dev/docs/index.html#entrydocs-augmentedreality-properties-canActivateAR
